src/libNetwork/RumorManager.cpp

```diff
@@ -157,49 +157,53 @@ bool RumorManager::Initialize(const std::vector<Peer>& peers,
 
 bool RumorManager::AddRumor(const RumorManager::RawBytes& message) {
   LOG_MARKER();
-  {
-    std::lock_guard<std::mutex> guard(m_continueRoundMutex);
-    if (!m_continueRound) {  // Seems logical error. Round should have started.
-      if (message.size() > 0) {
-        SHA2<HASH_TYPE::HASH_VARIANT_256> sha256;
-        sha256.Update(message);  // raw_message hash
+  if (message.size() > 0) {
+    SHA2<HASH_TYPE::HASH_VARIANT_256> sha256;
```

Comment:
can use hash utils function here

Reply:
These were been used in previous version. Which hash utils function are you referring?

Reply:
static const std::vector<unsigned char> BytesToHash(
      const std::vector<unsigned char>& vec) {
    SHA2<HASH_TYPE::HASH_VARIANT_256> sha2;

    sha2.Update(vec);
    const std::vector<unsigned char>& resVec = sha2.Finalize();

    return resVec;
  }